nengo_gui/static/side_menu.js

```diff
@@ -0,0 +1,189 @@
+ * upon page load
+ */
+
+Nengo.SideMenu = function(){
+  var self = this;
+  var file_browser_open = false
+  //Menu initially open
+  self.menu_open = true;
+
+  self.menu_width = document.getElementsByClassName("sidenav-container")[0].offsetWidth;
+
+  // Gathers all the menu tabs from the HTML for the Event Handlers
+  self.sub_menus = [];
+  self.tabs = $(".tab-content");
+  self.refresh();
+
+  //----EVENT HANDLERS----
+
+  //file_browser
+  $('#Open_file_button')[0].addEventListener('click', function () {
+      if (!$(this).hasClass('deactivated') && file_browser_open == false) {
+          file_browser_open = true;
+          self.file_browser();
+      }
+  });
+
+  //Modal Handlers
+  $('#Pdf_button')[0].addEventListener('click', function () {
+      self.pdf_modal();
+  });
+  $('#Download_button')[0].addEventListener('click', function () {
+      self.csv_modal();
+  });
+  $('#Minimap_button')[0].addEventListener('click', function () {
+      Nengo.netgraph.toggleMiniMap();
+  });
+  $('#Toggle_drag_drop')[0].addEventListener('click', function () {
+      self.toggle_display_tab("#Component_menu");
+  });
+
+  //Toggles Side Nav
+  $('.side_toggle').on('click',function(){
+      self.toggle_side_nav();
+  });
+
+    // Handles Accordions
+  var accord_num = document.getElementsByClassName('accordion-container').length+1;
+  for(var x = 1; x < accord_num; x++){
+    $('#accordion-container'+String(x)).find('.accordion-toggle').click(function(){
+
+      //Expand or collapse this panel
+      $(this).next().slideToggle();
+
+      //Hide the other panels
+      $(".accordion-content").not($(this).next()).slideUp();
+
+    });
+  }
+  //Initiate After Event Handlers
+  $('#Config_menu').click();
+  var h = $('.sidenav-container').height();
+  $('#filebrowser').height(h-48);
+
+};
+
+//** Finds all menu tabs and creates handlers **/
+Nengo.SideMenu.prototype.refresh = function(){
+    var self = this;
+    //Uninitiated
+    if(self.sub_menus.length == 0){
+        var menu_tabs = $(".menu_tab");
+        for(var x = 0; x < menu_tabs.length; x++){
+            var id = $(menu_tabs[x]).attr('id');
+            var clss = $(menu_tabs[x]).attr('class');
+            if(clss.indexOf("hidden") > 0){
+                $(self.tabs[x]).hide();
+                self.sub_menus.push({id:"#"+id, tab:self.tabs[x], hidden:true});
+            } else {
+                self.sub_menus.push({id:"#"+id, tab:self.tabs[x], hidden:false});
+            }
+        }
+    }
+    //Handles Menu tab switching
+    var counter = 0;
+    for(var x = 0; x < self.sub_menus.length; x++){
+        if(!self.sub_menus[x].hidden){
+            $(self.sub_menus[x].id).click(this.menu_tab_click(self.sub_menus[x].id,counter));
+            counter++;
+        }
+    }
+}
+
+//** Toggle show/hide of a given tab **/
+Nengo.SideMenu.prototype.toggle_display_tab = function(tab_id){
+    var self = this;
+    var index = 0;
+    while(index < self.sub_menus.length){
```

Comment:
Is this while-loop totally necessary? Could the mapping between tab_id and index be saved in a dictionary?